### PR TITLE
Fix #615: Rename channel_settings files to channel_subscriptions 

### LINKED ISSUE #615: Rename channel_settings files to channel_subscriptions
The term `settings` is more appropriately named `subscriptions`. If we are configuring any `settings` for a channel, the only option is to actually configure `subscriptions`. The task is to rename files, components, function names, .etc appropriately as subscriptions, not settings.

*Issue transferred from https://github.com/mattermost/mattermost-plugin-ee-jira/issues/30*

### PR DESCRIPTION
**Pull request by haardikdharma10**
#### Summary

This PR aims to solve #615 

#### Ticket Link

Issue: #615

### CODE REVIEW

**Review by jfrerich — APPROVED** [review]
Awesome!  Thank you, @haardikdharma10!

File: webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filter.tsx [hunk 1]
```diff
@@ -26,7 +26,7 @@ export type State = {
     error: string | null;
 }
 
-export default class ChannelSettingsFilter extends React.PureComponent<Props, State> {
+export default class ChannelSubscriptionsFilter extends React.PureComponent<Props, State> {
```
Comment (mickmister) [inline webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filter.tsx:29]:
```suggestion
export default class ChannelSubscriptionFilter extends React.PureComponent<Props, State> {
```

File: webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filter.tsx [hunk 2]
```diff
@@ -335,7 +335,7 @@ export default class ChannelSettingsFilter extends React.PureComponent<Props, St
     }
 }
 
-type EmptyChannelSettingsFilterProps = {
+type EmptyChannelSubscriptionsFilterProps = {
```
Comment (mickmister) [inline webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filter.tsx:338]:
```suggestion
type EmptyChannelSubscriptionFilterProps = {
```

File: webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filter.tsx [hunk 3]
```diff
@@ -344,7 +344,7 @@ type EmptyChannelSettingsFilterProps = {
     cancelAdd: () => void;
 };
 
-export function EmptyChannelSettingsFilter(props: EmptyChannelSettingsFilterProps): JSX.Element {
+export function EmptyChannelSubscriptionsFilter(props: EmptyChannelSubscriptionsFilterProps): JSX.Element {
```
Comment (mickmister) [inline webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filter.tsx:347]:
```suggestion
export function EmptyChannelSubscriptionFilter(props: EmptyChannelSubscriptionFilterProps): JSX.Element {
```

File: webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filters.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ type State = {
     showCreateRow: boolean;
 };
 
-export default class ChannelSettingsFilters extends React.PureComponent<Props, State> {
+export default class ChannelSubscriptionsFilters extends React.PureComponent<Props, State> {
```
Comment (mickmister) [inline webapp/src/components/modals/channel_subscriptions/channel_subscriptions_filters.tsx:25]:
```suggestion
export default class ChannelSubscriptionFilters extends React.PureComponent<Props, State> {
```

File: webapp/src/components/modals/channel_subscriptions/edit_channel_subscriptions.tsx [hunk 5]
```diff
@@ -71,7 +71,7 @@ export type State = {
     conflictingError: string | null;
 };
 
-export default class EditChannelSettings extends PureComponent<Props, State> {
+export default class EditChannelSubscriptions extends PureComponent<Props, State> {
```
Comment (mickmister) [inline webapp/src/components/modals/channel_subscriptions/edit_channel_subscriptions.tsx:74]:
```suggestion
export default class EditChannelSubscription extends PureComponent<Props, State> {
```

**Review by mickmister — CHANGES_REQUESTED** [review]
Great work @haardikdharma10! Thanks for tackling this task

I have a request to rename some of the plural `ChannelSubscriptions` to `ChannelSubscription`, for cases where we are only working with one subscription. Other than this, the PR looks perfect :+1:

**Comment by haardikdharma10** [comment]
@mickmister, yes, that makes sense. Applied your suggestions. Thanks for the review:)

**Comment by haardikdharma10** [comment]
Hi @mickmister, here: 

```
import {ChannelSubscription, ChannelSubscriptionFilters, ReactSelectOption, FilterValue, IssueMetadata} from 'types/model';

import ChannelSubscriptionsFilters from './channel_subscriptions_filters';
```

When I try to change `ChannelSubscriptionsFilters` to  `ChannelSubscriptionFilters` in the second import statement (as suggested by you) it gives me an error as follows:

Screenshot 2021-01-14 at 12 57 43 AM

as it is duplicated from the first import statement, which is leading to coverage errors if not changed. What should be done in this case?

**Comment by mickmister** [comment]
@haardikdharma10 It looks like there's a conflict between the name of the type in `types/model.ts` called `ChannelSubscriptionFilters`, and the React component's name. I think this solution should work:

```ts
import {ChannelSubscriptionFilters as ChannelSubscriptionFiltersModel} from 'types/model';

import ChannelSubscriptionsFilters from './channel_subscriptions_filters';
```

**Comment by mickmister** [comment]
@haardikdharma10 Can we also rename the files to singular like
`channel_subscriptions_filter -> channel_subscription_filter`

**Comment by haardikdharma10** [comment]
@mickmister, all test cases pass and this PR is ready for re-review 😅

**Review by mickmister — APPROVED** [review]
Awesome work @haardikdharma10! This is a great improvement to the codebase :+1:

**Comment by haardikdharma10** [comment]
ping @DHaussermann, can you please review this PR and let us know if this is good to go!? Thanks :)

**Review by DHaussermann — APPROVED** [review]
Tested and passed
Regression tested subscriptions functionality for this change
- Performed Create/Update/Delete testing of subscriptions
- Ensured channel notices still work when subscriptions are edited 
- Subscription list modal and create/edit modal behave normally 
- Confirmed subscriptions are still functional and Projects, Event, additional filters etc... get updated properly and subscriptions get delivered accordingly
- Subscribe list still works as expected   

LGTM!

No issues found. 
No new release tests needed to cover this. 

@haardikdharma10 huge thanks for this enhancement. Much appreciated! My apologies for the delay in testing this change.

**Pull request merged**